### ISSUE TITLE
The beta8 `project.json` commands should be rock-solid for end users and aligned with docs.asp.net for beta8

### ISSUE THREAD

**Issue by peterblazejewicz** [OmniSharp/generator-aspnet#422]
moved from: https://github.com/OmniSharp/generator-aspnet/issues/386#issuecomment-148500445

There is a problem with `project.json` commands for web content type of applications.
We were adviced (after some objections) to use `web` as new, all  `kestrel` command - similar to solution used in VS templates, but the official DNX samples that are on aspnet/home still use separate:
https://github.com/aspnet/Home/blob/dev/samples/1.0.0-beta8/HelloWeb/project.json#L21
https://github.com/aspnet/Home/blob/dev/samples/1.0.0-beta8/HelloMvc/project.json#L20-L21

We should decide which version to support and update `project.json` accordingly. We should ask/reach people involved in release, to have a firm stance on this.

My stance is to update `project.json` to use both, the same way as in aspnet/home
Thanks!

**Comment by david-driscoll** [comment]
I would agree.

**Comment by scottaddie** [comment]
I also agree with both @peterblazejewicz and @david-driscoll.

**Comment by spboyer** [comment]
I made the suggestion that we add `web` and `kestrel` in the commands both pointing to `Microsoft.AspNet.Server.Kestrel` in another thread.

**Comment by peterblazejewicz** [comment]
I've found yesterday that my argument about docs.asp.net is no longer valid. The authors of docs started to use unified "web":
https://github.com/aspnet/Docs/blob/master/common/samples/WebApplication1/src/WebApplication1/project.json#L33
https://github.com/aspnet/Docs/blob/9129a71774e1f0385c704360c5b28dcae0c64ec9/mvc/getting-started/first-web-api/sample/src/TodoApi/project.json#L16
They have added many commits starting two days ago that updates docs:
https://github.com/aspnet/Docs/commits/master
The aspnet/home samples are still behind, but there is a visible direction in changes.
Can we have someone from MS to give a hint?

**Comment by Kagamine** [comment]
You can still use Microsoft.AspNet.Hosting in beta8, or later.
Microsoft.AspNet.Hosting is a console application, and here is the source: https://github.com/aspnet/Hosting/blob/dev/src/Microsoft.AspNet.Hosting

The kestrel depend on the Hosting, you can run kestrel without hosting. You can see it in the following code:

```
using System;
using System.Linq;

namespace Microsoft.AspNet.Server.Kestrel
{
    public class Program
    {
        public static void Main(string[] args)
        {
            var mergedArgs = new[] { "--server", "Microsoft.AspNet.Server.Kestrel" }.Concat(args).ToArray();
            Microsoft.AspNet.Hosting.Program.Main(mergedArgs);
        }
    }
}
```

This is Microsoft.AspNet.Server.Kestrel entry point. it called hosting.

**Comment by sayedihashimi** [comment]
Kestrel is now our primary hosting mechanism, because of this the old host which was named `web` is no longer needed. It was windows specific as well.

We have moved to use `web` instead of `kestrel`. It will take a while to get everything updated, but the changes are in progress.

In addition to this, when you open a project in Visual Studio if it doesn't have a `web` command then it will be created automatically. Additionally if it points to the old windows specific host, it will be updated to use Kestrel. See this related thread https://github.com/aspnet/Tooling/issues/231. Tomorrow I'll create an announcement that the `web` command is required for web projects starting in `beta8`.

**Comment by Kagamine** [comment]
I saw the WebListener is still in development. Is there any plan to cancel it?

**Comment by sayedihashimi** [comment]
@kagamine I wouldn't know, but I haven't heard of any such plans.

**Comment by Kagamine** [comment]
I think the signalr v3 has not support the kestrel, cancel it will block the signalr team.

**Comment by sayedihashimi** [comment]
Yeah there are likely to be many more reasons like that.

**Comment by Kagamine** [comment]
Hah, I have a question, Does the signalr have to support kestrel or kestrel support signalr？

**Comment by sayedihashimi** [comment]
I don't know, you can ask on the signalr repo.

**Comment by peterblazejewicz** [comment]
> Tomorrow I'll create an announcement that the web command is required for web projects starting in beta8.

@sayedihashimi 
Yes, I think this should be escalated to a topic shared via https://github.com/aspnet/Announcements
Thanks!

**Comment by peterblazejewicz** [comment]
We are shutting this repo down, it's been replaced with `dotnet new`.

Thanks!

**Comment by sayedihashimi** [comment]
We are shutting this repo down, it's been replaced with `dotnet new`.
